facturacion_mexico/facturacion_fiscal/cancellation_state.py

```
@@ -0,0 +1,171 @@
+	"""Derivar el texto del Select `cancellation_reason` desde el código de motivo solicitado.
+
+	Se deriva por prefijo contra las opciones reales del DocType (evita drift con descripciones).
+	Si no hay motivo o no matchea ninguna opción, devuelve "" (fail-soft: no bloquea la transición).
+	"""
+	motivo = (motivo or "").strip()
+	if not motivo:
+		return ""
+	field = frappe.get_meta("Factura Fiscal Mexico").get_field("cancellation_reason")
+	opts = [(o or "").strip() for o in (field.options or "").split("\n") if (o or "").strip()]
+	for o in opts:
+		if o.startswith(f"{motivo} -") or o.startswith(f"{motivo} "):
```

Ruff (0.15.18)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)